app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRPresenter.java

```diff
@@ -0,0 +1,116 @@
+package org.fossasia.openevent.app.qrscan;
+
+import android.util.SparseArray;
+
+import com.google.android.gms.vision.barcode.Barcode;
+
+import org.fossasia.openevent.app.data.contract.IEventDataRepository;
+import org.fossasia.openevent.app.data.models.Attendee;
+import org.fossasia.openevent.app.qrscan.contract.IScanQRPresenter;
+import org.fossasia.openevent.app.qrscan.contract.IScanQRView;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import io.reactivex.Observable;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+import io.reactivex.subjects.PublishSubject;
+
+public class ScanQRPresenter implements IScanQRPresenter {
+
+    private long eventId;
+
+    private IScanQRView scanQRView;
+    private IEventDataRepository eventDataRepository;
+    private List<Attendee> attendees = new ArrayList<>();
+
+    private PublishSubject<Boolean> detect = PublishSubject.create();
+    private PublishSubject<String> data = PublishSubject.create();
+
+    public ScanQRPresenter(long eventId, IScanQRView scanQRView, IEventDataRepository eventDataRepository) {
+        this.eventId = eventId;
+        this.scanQRView = scanQRView;
+        this.eventDataRepository = eventDataRepository;
+
+        detect.distinctUntilChanged()
+            .debounce(150, TimeUnit.MILLISECONDS)
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe(receiving -> scanQRView.showInView(!receiving));
+
+        data.distinctUntilChanged()
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe(this::processBarcode);
+    }
+
+    private void processBarcode(String barcode) {
+        scanQRView.showData(barcode);
+
+        Observable.fromIterable(attendees)
+            .filter(attendee -> (attendee.getOrder().getIdentifier() + "-" + attendee.getId()).equals(barcode))
```

Comment:
But I am thinking if we bring all the data related functionality to a single class it will be handy to pass that single class object to any presenter which will handle all the data logic.